### PR TITLE
Disable analyzers only when building in VS

### PR DESCRIPTION
**Pull request by captainsafia**
This check was causing analyzers not to run when building on the CLI. Updating to only apply for VS builds.

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Caught one!
> src/Servers/Kestrel/Core/src/MinDataRate.cs(47,32): error RS0016: (NETCORE_ENGINEERING_TELEMETRY=Build) Symbol 'ToString' is not part of the declared API.

**Review by halter73 — APPROVED** [review]
I think we just add MinDataRate.ToString() to the PublicApi.Unshipped so we don't have to revert https://github.com/dotnet/aspnetcore/pull/34680. @davidfowl

I guess we can mark this PR `api-ready-for-review` and look at it after it's merged. I can't imagine there being any pushback.

**Review by JunTaoLuo — APPROVED** [review]
I think we should get this in asap so we don't continue to "accidentally modify APIs"

**Comment by dougbu** [comment]
> I think we should get this in asap so we don't continue to "accidentally modify APIs"

Agreed. Fix the missing `ToString()` method in VS (16.x, not 17.x 😺) and be done w/ it. Use admin privs to shorten the window for busted stuff

**Review by JamesNK — APPROVED** [review]

File: Directory.Build.props
Comment (pranavkm) [inline Directory.Build.props:18]: Could someone remind me why we did this? Is this because of all of the failures to load the analyzer dll?

**Comment by dougbu** [comment]
@pranavkm please see https://teams.microsoft.com/l/message/19:8d76f8f67d8e4237b9f0b87235ec870e@thread.skype/1626473819862?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=fdff90ed-0b3b-4caa-a30a-efb4dd47665f&parentMessageId=1626473819862&teamName=DotNet&channelName=ASP.NET%20Build&createdTime=1626473819862

File: Directory.Build.props
Comment (halter73) [reply Directory.Build.props:18]: Yes. https://github.com/dotnet/aspnetcore/pull/34664 And I didn't verify the analyzers still ran on builds outside of VS 😞

**Review by dougbu — APPROVED** [review]
@dotnet/aspnet-build please double-check this then merge w/o waiting for CI. This should stop the bleeding in subsequent PRs though we need to be careful w/ PRs already marked as successful.

**Comment by dougbu** [comment]
@dotnet/aspnet-build never mind on the admin squash. Build failures are real and look like additional API changes as well as a new non-MIT header. ☹️

> And I didn't verify the analyzers still ran on builds outside of VS 😞

@halter73 counter-intuitively, `$(VisualStudioVersion)` is `17.0` when building in `dotnet msbuild` with the current SDK. No way you could have predicted that.

@dsplaisted @rainersigwald @wli3 is that property setting coming from the SDK itself❔ If not, where❔ I expected no property setting outside desktop `msbuild`.

----

@captainsafia @BrennanConroy @JunTaoLuo and others playing in this PR: Have a look at other `Condition`s using `$(VisualStudioVersion)`, especially the one in eng/targets/CSharp.Common.targets. Ignore the `*.vcxproj` files and src/Servers/IIS/build/Build.Settings because they're used only in _native_ builds whether inside VS or not.

**Comment by dougbu** [comment]
/fyi @davidfowl one non-MIT header (in WebApplicationServiceCollection.cs) comes from d434b1c43529.

/fyi @javiercn and @JunTaoLuo the other non-MIT header (in ManifestStaticWebAssetsFileProviderTests.cs) comes from 014df779b4d0. Downside of the CI not enforcing comments on the PR&hellip;

**Comment by captainsafia** [comment]
> Have a look at other Conditions using $(VisualStudioVersion), especially the one in eng/targets/CSharp.Common.targets

Since this is part of the same workaround to disable loading analyzers in certain versions of VS, I think it's fine to make this conditional on the build happening in VS too. Not sure if it'll actually cause any tangible warnings in the CI but we'll see.

**Review by TanayParikh — APPROVED** [review]

**Comment by captainsafia** [comment]
@dotnet/aspnet-build Ooooook. Looks like the build legs are starting to go green. I think this is good to merge at your discretion.

File: eng/targets/CSharp.Common.targets
Comment (dougbu) [inline eng/targets/CSharp.Common.targets:39]: same nit:
```suggestion
    <EnforceCodeStyleInBuild Condition=" '$(MSBuildRuntimeType)' != 'Core' AND '$(VisualStudioVersion)' &lt; '17.0' ">false</EnforceCodeStyleInBuild>
```

Separately have we confirmed `<` works with strings that look like floating-point numbers❔ @rainersigwald❔

**Review by dougbu — APPROVED** [review]
Let's hope this beats other API changes and file additions 😀

**Comment by dougbu** [comment]
Ick Components.E2ETest has not been reliable recently ☹️ /fyi @JunTaoLuo if you're still wrapping up blops

**Comment by dougbu** [comment]
@captainsafia please get this in w/ your admin privs if the test failure is all that goes wrong in this build. No need to wait for Helix jobs for example and don't bother retrying the failed 'Test: Windows Server 2016 x64' job.

**Comment by captainsafia** [comment]
I think this is safe to merge. There's a few build legs that are still queued to run (macOS arm64, Linux arm64) but I think it is safe to merge.

I'll follow up to feedback and any potential fallout as part of my build-ops rotation.

File: eng/targets/CSharp.Common.targets
Comment (rainersigwald) [reply eng/targets/CSharp.Common.targets:39]: it does but consider using the newish [version comparison property functions](https://docs.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2019#msbuild-version-comparison-functions)

**Pull request merged**
